canonical-web-and-design/maas-ui
# Add interfaces table to KVM compose form.

## Done

- Added interfaces table to KVM compose form, which is used to construct the interface constraints string.
- Updated warning styling for cores and memory fields

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM details page of a KVM and open the Compose dropdown.
- Check that there is an Interfaces section.
- Click "Define (optional)" and define some interface constraints.
- Check that the list of subnets is filtered depending on the selected space.
- Check that the values for fabric, vlan and PXE update correctly.
- Compose the machine and wait a minute or so for MAAS to set it up.
- Check that the newly created machine has interfaces as defined in the KVM compose form.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2042

## Screenshot
### Undefined interface
![0 0 0 0_8400_MAAS_r_kvm_185 (7)](https://user-images.githubusercontent.com/25733845/88360248-64615b80-cdb8-11ea-9331-e945e64bb5f5.png)



### Defined interface
![0 0 0 0_8400_MAAS_r_kvm_185 (8)](https://user-images.githubusercontent.com/25733845/88360262-67f4e280-cdb8-11ea-87eb-f1797ec00fc6.png)



### Mobile view
![0 0 0 0_8400_MAAS_r_kvm_185 (12)](https://user-images.githubusercontent.com/25733845/88364932-cd040480-cdc7-11ea-8c60-267123d1e9ca.png)
